map.js

```diff
@@ -46,51 +46,90 @@ legend.onAdd = function() {
 };
 legend.addTo(map);
 
+/*
+fetch (states10mSource)
+  .then(response => response.json())
+  .then(function(json) {
+      L.vectorGrid.slicer(json, {
+        rendererFactory: L.canvas.tile,
+        vectorTileLayerStyles: {
+          sliced: function (properties, zoom) {
+            return stateStyle(properties);
+          }
+        }
+      }).addTo(map);
+  });
+*/
 const statesLayer = new ZoomShowHide().addTo(map);
 
 const ZOOM_THRESHOLD = 5;
 
-const states50m = L.geoJSON(null, {
-  style: stateStyle
-});
-statesLayer.addLayer(states50m);
+let states50m = '';
```

Comment:
You can try defining the layer as an empty `L.vectorGrid`, but you'd have to find out how to add the data to the layer later (not redefining it with `L.vectorGrid.slicer`.

Or we can probably drop the global definition altogether and just instantiate the layer directly in the `fetch.then` method.

Reply:
You would also need to get the `states50m` object reference back from the `statesLayer` group for the call on line 115, so I think it's ok to keep the global reference (but it's better to define it without value like this: `let states50m;`.

Reply:
Fixed. `let states50m;` declaration used.